### DISCUSSION TITLE
Auto open empty dir

### DISCUSSION THREAD

**Discussion by Tylopilus** [nvim-neo-tree/neo-tree.nvim#1452] [Q&A] [answered]
When working in java projects you often end up with empty folders. Is there an option to automatically open all sub folders if the folder is empty?

Basically if I have a structure of:
* Folder 1
  * Folder 2
    * Folder 3
      * Folder 4
        * main.file

If I open Folder 1 it should automatically open down to folder 4 to expose main.file as folder 4 is the first non-empty folder (has more than 1 subfolder as a child) 

This is nicely handled in vscode and I would like to have the same functionality in neo-tree. I am already using `group_empty_dirs = true` but that is only a visual change and still requires to open all empty folders.

**Comment by pysan3** [comment, marked as answer]
Yes it is `filesystem.scan_mode = "deep"`. Please read `defaults.lua` for better explanation as I'm on my phone right now. 

https://github.com/nvim-neo-tree/neo-tree.nvim/blob/main/lua/neo-tree/defaults.lua

**Reply by Tylopilus** [reply]
Thanks so much! that does indeed work as expected but apparently `group_empty_dirs=true` must be set for it to take effect. Thanks for the quick help!

**Comment by Tylopilus** [comment]
Thanks. This is the solution I was after.
